tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,494 @@
-        assertNotNull(result.get(GROUP),
-            "Group was deleted successfully, but it shouldn't have been. Result was:(" + result + ")");
-        assertTrue(result.size() == 1 && result.containsKey(GROUP) && result.get(GROUP).getCause() instanceof GroupNotEmptyException,
-            "The expected error (" + Errors.NON_EMPTY_GROUP + ") was not detected while deleting consumer group. Result was:(" + result + ")");
+    @ClusterTest
+    public void testDeleteNonExistingGroup() throws Exception {
+        try (Admin admin = cluster.createAdminClient()) {
+            createAndAwaitTestTopic(admin);
+            String missingGroup = "missing.group";
+            // note the group to be deleted is a different (non-existing) group
+            String[] cgcArgs = new String[]{"--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", missingGroup};
+            ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+            Map<String, Throwable> result = service.deleteGroups();
+            assertTrue(result.size() == 1 && result.containsKey(missingGroup) && result.get(missingGroup).getCause() instanceof GroupIdNotFoundException,
+                    "The expected error (" + Errors.GROUP_ID_NOT_FOUND + ") was not detected while deleting consumer group");
+        }
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testDeleteCmdEmptyGroup(String quorum) throws Exception {
-        createOffsetsTopic(listenerName(), new Properties());
-
-        // run one consumer in the group
-        ConsumerGroupExecutor executor = addConsumerGroupExecutor(1);
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
-
-        TestUtils.waitForCondition(
-            () -> service.listConsumerGroups().contains(GROUP) && Objects.equals(service.collectGroupState(GROUP).state, "Stable"),
-            "The group did not initialize as expected."
-        );
-
-        executor.shutdown();
-
-        TestUtils.waitForCondition(
-            () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
-            "The group did not become empty as expected."
-        );
-
-        String output = ToolsTestUtils.grabConsoleOutput(service::deleteGroups);
-        assertTrue(output.contains("Deletion of requested consumer groups ('" + GROUP + "') was successful."),
-            "The consumer group could not be deleted as expected");
+    @ClusterTest
+    public void testDeleteCmdNonEmptyGroup() throws Exception {
+        try (
+                Admin admin = cluster.createAdminClient();
+                ConsumerGroupExecutor consumerGroupExecutor = buildConsumerGroupExecutor(GROUP)
+        ) {
+            createAndAwaitTestTopic(admin);
```

Comment:
Kafka will auto-create the topic when the topic is nonexistent, so you don't need to call `createAndAwaitTestTopic`